ddtrace/appsec/iast/loader.py

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env python3
+
+from ddtrace.internal.logger import get_logger
+from ddtrace.internal.module import ModuleWatchdog
+
+from ._ast.ast_patching import _should_iast_patch
+from ._ast.ast_patching import astpatch_source
+from ._util import _is_iast_enabled
+
+
+log = get_logger(__name__)
+
+
+IS_IAST_ENABLED = _is_iast_enabled()
+
+
+def _exec_iast_patched_module(module_watchdog, module):
+    patched_source = None
+    if IS_IAST_ENABLED:
+        log.debug("IAST enabled")
+        module_path, patched_source = astpatch_source(module.__name__, None)
+
+    if patched_source:
+        # Patched source is executed instead of original module
+        compiled_code = compile(patched_source, module_path, "exec")
+        exec(compiled_code, module.__dict__)
+    else:
+        module_watchdog.loader.exec_module(module)
```

Comment:
Look at all the different branches that return ``"", ""`` here: https://github.com/DataDog/dd-trace-py/blob/1.x/ddtrace/appsec/iast/_ast/ast_patching.py#L59

Reply:
Would it be possible to raise an exception here that would be handled in the `_exec_module` from the `ddtrace.internal.module` to avoid manually calling from the loader? Or could that be a potential issue if we want to pipe multiple transformers in the future? I'd really love if we could get rid of this line 🙂

Reply:
In fact, thinking a bit more about this made me realise that the current design is flawed 🙁 . We cannot execute a module here, either with `exec` or with `loader.exec_module`. If every pre-exec hook we register does that, then we would be executing a module more than once, which is wrong by Python standards. So we either execute one hook, or we go back to the idea of having hook take the AST and return a modified AST that is then executed by `_exec_module` in `ddtrace.internal.module`.

Reply:
OK, just to unblock this PR, how about checking there is a single hook for the module, and tackle the chained transformations in a future PR? I propose this because we don't need to hurry with the chained transformations AFAIK

Reply:
Yup. To give more context, we don't expect to have more than one transformation for the IAST part, of course things could change in the future but for the first version I think limiting (and checking that only run) one hook per module should be fine.